win32ss/gdi/ntgdi/freetype.c

```diff
@@ -3226,32 +3273,15 @@ ftGdiGetGlyphOutline(
     IntLockFreeType();
 
     /* Scaling transform */
-    /*if (aveWidth)*/
+    if (widthRatio != 1.0)
     {
+        FT_Matrix scaleMat;
+        scaleMat.xx = FT_FixedFromFloat(widthRatio);
+        scaleMat.xy = 0;
+        scaleMat.yx = 0;
+        scaleMat.yy = FT_FixedFromFloat(1.0);
 
-        FT_Matrix ftmatrix;
-        FLOATOBJ efTemp;
-
-        PMATRIX pmx = DC_pmxWorldToDevice(dc);
-
-        /* Create a freetype matrix, by converting to 16.16 fixpoint format */
-        efTemp = pmx->efM11;
-        FLOATOBJ_MulLong(&efTemp, 0x00010000);
-        ftmatrix.xx = FLOATOBJ_GetLong(&efTemp);
-
-        efTemp = pmx->efM12;
-        FLOATOBJ_MulLong(&efTemp, 0x00010000);
-        ftmatrix.xy = FLOATOBJ_GetLong(&efTemp);
-
-        efTemp = pmx->efM21;
-        FLOATOBJ_MulLong(&efTemp, 0x00010000);
-        ftmatrix.yx = FLOATOBJ_GetLong(&efTemp);
-
-        efTemp = pmx->efM22;
-        FLOATOBJ_MulLong(&efTemp, 0x00010000);
-        ftmatrix.yy = FLOATOBJ_GetLong(&efTemp);
-
-        FT_Matrix_Multiply(&ftmatrix, &transMat);
+        FT_Matrix_Multiply(&scaleMat, &transMat);
```

Comment:
Shouldn't we take also into account any `DC_pmxWorldToDevice(dc);` ?

Reply:
Yes, we should do, but I don't know how to do.

Reply:
OK, Fixed.